services/movie_session.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> list[MovieSession]:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> MovieSession:
+    movie_session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        movie_session.show_time = show_time
+    if movie_id:
+        movie_session.movie_id = movie_id
+    if cinema_hall_id:
+        movie_session.cinema_hall_id = cinema_hall_id
+    movie_session.save()
+    return movie_session
+
+
+def delete_movie_session_by_id(movie_session_id: int) -> None:
+    MovieSession.objects.get(id=movie_session_id).delete()
```

Comment:
The `delete_movie_session_by_id` function uses `MovieSession.objects.get(id=movie_session_id).delete()`, which will raise a `MovieSession.DoesNotExist` exception if the session with the given ID is not found. Consider handling this exception to provide a more user-friendly error message or behavior.